packages/xod-arduino/src/transpiler.js

```diff
@@ -91,10 +91,16 @@ const getPatchByNodeId = def(
 );
 
 const toposortProject = def(
-  'toposortProject :: PatchPath -> Project -> Either Error Project',
+  'toposortProject :: PatchPath -> Project -> Either Error ProjectWithNodeIdsMap',
```

Comment:
The `ProjectWithNodeIdsMap` looks wild a bit to me. Evolution would lead to `ProjectWithNodeIdsMapLinksExpandedAndDuplicationsSquashed`.

Wouldn’t it be more clear to have a simple additional field like `originalId` right inside any `TNode`?